community/x402-template/lib/x402-verification.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * x402 Payment Verification
+ *
+ * Handles payment verification via facilitator
+ */
+
+import transactionStorage from './transaction-storage'
+import { X402_CONFIG } from './x402-config'
+
+interface PaymentPayload {
+  signature: string
+  from: string
+  to: string
+  amount: string
+  token: string
+}
+
+interface Payment {
+  x402Version: number
+  scheme: string
+  network: string
+  payload: PaymentPayload
+}
+
+interface VerificationResult {
+  isValid: boolean
+  signature?: string
+  reason?: string
+}
+
+export async function verifyPayment(payment: Payment, resource: string): Promise<VerificationResult> {
+  try {
+    if (!payment.payload || !payment.payload.signature) {
+      return { isValid: false, reason: 'Invalid payment structure' }
+    }
+
+    const { signature, from, to, amount } = payment.payload
+
+    if (await transactionStorage.has(signature)) {
+      return { isValid: true, signature }
+    }
+
+    const paymentRequirements = {
+      scheme: 'exact',
+      network: 'solana-devnet',
```

Comment:
let's avoid hard coding stuff, we should be able to use the template quickly into production without scouting for hard coded values in the repo.

move to env files